### ISSUE TITLE
Bug: no-undef object key in optional call

### ISSUE THREAD

**Issue by helixbass** [helixbass/eslint-plugin-coffee#13]
This is triggering `no-undef` for `x` somehow:
```
@?.get x: 10
```

**Comment by helixbass** [comment]
Another (presumably related) example:
```
do (b = (c) -> c) -> a: 3
?.a
```

**Comment by helixbass** [comment]
Maybe related:

This is flagging `type`:
```
class A
  category:  (type: 'string')
```
